plugins/modules/system/homectl.py

```diff
@@ -0,0 +1,618 @@
+            name=dict(type='str', required=True, aliases=['user', 'username']),
+            password=dict(type='str', no_log=True),
+            storage=dict(type='str', choices=['classic', 'luks', 'directory', 'subvolume', 'fscrypt', 'cifs']),
+            disksize=dict(type='str'),
+            resize=dict(type='bool', default=False),
+            realname=dict(type='str', aliases=['comment']),
+            realm=dict(type='str'),
+            email=dict(type='str'),
+            location=dict(type='str'),
+            iconname=dict(type='str'),
+            homedir=dict(type='path'),
+            imagepath=dict(type='path'),
+            uid=dict(type='int'),
+            gid=dict(type='int'),
+            umask=dict(type='int'),
+            environment=dict(type='str', aliases=['setenv']),
+            timezone=dict(type='str'),
+            memberof=dict(type='str', aliases=['groups']),
+            skeleton=dict(type='path', aliases=['skel']),
+            shell=dict(type='str'),
+            locked=dict(type='bool'),
+            passwordhint=dict(type='str', no_log=True),
+            sshkeys=dict(type='str', no_log=True),
+            language=dict(type='str'),
+            notbefore=dict(type='int'),
+            notafter=dict(type='int'),
+            mountopts=dict(type='str'),
+        ),
+        supports_check_mode=True,
+
+        required_if=[
+            ('state', 'present', ['password']),
+            ('resize', True, ['disksize']),
+        ]
+    )
+
+    homectl = Homectl(module)
+    homectl.result['state'] = homectl.state
+
+    # First we need to make sure homed service is active
+    if not homectl.homed_service_active():
+        module.fail_json(msg="systemd-homed.service is not active")
+
+    # handle removing user
+    if homectl.state == 'absent':
+        if homectl.user_exists():
+            if module.check_mode:
+                module.exit_json(changed=True)
+            rc, stdout, stderr = homectl.remove_user()
+            if rc != 0:
+                module.fail_json(name=homectl.name, msg=stderr, rc=rc)
+            homectl.result['rc'] = rc
+            homectl.result['msg'] = "User %s removed!" % homectl.name
+        else:
+            homectl.result['changed'] = False
+            homectl.result['msg'] = "User does not exist!"
+
+    # Handle adding a user
+    if homectl.state == 'present':
+        if not homectl.user_exists():
+            if module.check_mode:
+                module.exit_json(changed=True)
+            rc, stdout, stderr = homectl.create_user()
+            if rc != 0:
+                module.fail_json(name=homectl.name, msg=stderr, rc=rc)
+            user_metadata = json.loads(homectl.get_user_metadata())
+            homectl.result['data'] = user_metadata
+            homectl.result['rc'] = rc
+            homectl.result['msg'] = "User %s created!" % homectl.name
+        else:
+            if module.check_mode:
+                module.exit_json(changed=True)
```

Comment:
This looks wrong. At least the change detection should be run before determining that `changed=True`.

Reply:
I changed the logic, let me know if this seems better.

The reason I have all the  'self.result['changed'] = True'  is homed will overwrite any change to user record when using `--identity` so these checks ensure that what the user has configured doesn't get edited/cleared unless they are calling to change it with one of the module options. Let me know if this makes sense or I can try and explain further.